### PR TITLE
chore: add deploy workflow

### PR DESCRIPTION
**Pull request by ashwinkumar6**
#### Description of changes
Add a callable deploy workflow which would
-  publish the package on npm
- create a tagged release on gtihub with the release notes 

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are [changed or added](https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
 		"unlink-all": "lerna exec --no-bail --parallel -- yarn unlink; exit 0",
 		"publish:main": "lerna publish --canary --force-publish \"*\" --yes --dist-tag=unstable --preid=unstable${PREID_HASH_SUFFIX} --exact --no-verify-access",
 		"publish:next": "lerna publish --canary --force-publish \"*\" --yes --dist-tag=next --preid=next${PREID_HASH_SUFFIX} --exact --no-verify-access",
-		"publish:release": "lerna publish --conventional-commits --yes --message 'chore(release): Publish [ci skip]' --no-verify-access",
+		"publish:release": "lerna publish --conventional-commits --create-release github --yes --message 'chore(release): Publish [ci skip]' --no-verify-access",
```
Comment (stocaaro) [inline package.json]: Lets remove this for now.

File: .github/workflows/callable-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: 'Publish package to npm and create release on github'
+
+on:
+  workflow_call:
+
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GITHUB_EMAIL: test@email.com
+  GITHUB_USER: test-user
```
Comment (stocaaro) [inline .github/workflows/callable-deploy.yml:14]: Please add a TODO comment to get this from config/variables

File: .github/workflows/callable-deploy.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  BRANCH: release
```
Comment (stocaaro) [inline .github/workflows/callable-deploy.yml]: This should be an input from the calling workflow

**Review by stocaaro — COMMENTED** [review]
Looks good once the recommended changes are in.

File: package.json (hunk 1 above)
Comment (ashwinkumar6) [reply package.json]: sure, we can add this feature later on in a different PR

File: .github/workflows/callable-deploy.yml (hunk 2 above)
Comment (ashwinkumar6) [reply .github/workflows/callable-deploy.yml:14]: Added

File: .github/workflows/callable-deploy.yml (hunk 3 above)
Comment (ashwinkumar6) [reply .github/workflows/callable-deploy.yml]: Updated

**Review by stocaaro — APPROVED** [review]

**Pull request merged**